src/test/java/org/openrewrite/kotlin/tree/AnnotationTest.java

```diff
@@ -82,4 +84,122 @@ void jvmNameAnnotation() {
           )
         );
     }
+
+    @Test
+    @Issue("https://github.com/openrewrite/rewrite-kotlin/issues/105")
+    void fullyQualifiedCustomAnnotation() {
+        rewriteRun(
+          java(
+            """
+              package com.example.annotations;
+                            
+              import java.lang.annotation.Target;
+              import static java.lang.annotation.ElementType.*;
+                            
+              @Target({TYPE, FIELD, METHOD, PARAMETER, CONSTRUCTOR, LOCAL_VARIABLE})
+              public @interface JavaAnnotationToRemove {
+              }
+              """
+          ),
+          // Works
+          kotlin(
+            """
+              import com.example.annotations.JavaAnnotationToRemove
+
+              @JavaAnnotationToRemove
+              class A {
+              }
+              """),
+          // Fails, despite being valid Kotlin code
+          kotlin(
+            """
+              @com.example.annotations.JavaAnnotationToRemove
+              class A {
+              }
+              """)
+        );
+    }
+
+    @Test
+    @Issue("https://github.com/openrewrite/rewrite-kotlin/issues/105")
+    void fullyQualifieJavaLangAnnotation() {
+        rewriteRun(
+          // Works, Deprecated is defined in java.lang package so no import needed
+          kotlin(
+            """
+              @Deprecated
```

Comment:
In my IDE this resolves to `kotlin.Deprecated` (missing the required message argument), not `java.lang.Deprecated`. Only when I add `import java.lang.Deprecated` does this resolve to the Java type.

We likely should update the test to either add that import or required message argument.